Do not copy logs and cache into docker image

comstud opened the pull request:
The cache/logs shouldn't be copied back into new docker image builds, which is what happens when you attach these as volumes, run Golbat for a while, and then re-build the image. This was shoving ~175M of my logs to the docker daemon during build and copying into the image!

I think it only makes sense for people to attach these as volumes at runtime, and doing anything with them at image build time could just end up being an expensive build time no-op.

But even if not, if people do not attach these as volumes, they are always going to be empty on the host side such that there's nothing to copy into the image. But we still need to create the directories.

If you've compiled outside of docker, this also stops the sending of the 130M golbat binary through to the docker daemon when you next build, also.

na-ji commented:
I don't agree with this change for those reasons:
- Without those folders, Golbat will crash. The advantage of the current implementation is that they are optionals. Since the image is based on a distroless image, mkdir isn't available so that's why I used copy.
- Those folders are always empty on the CI so it's not really an issue for this kind of environment.

I do agree however that the content of the folders shouldn't be copied. If you can make it so the folders are still created by default without their content, that would be better.

jfberry commented:
I guess the difference here is the CI build vs a local build… nice to find a solution to both!

comstud commented:
> I don't agree with this change for those reasons:
> 
> * Without those folders, Golbat will crash. The advantage of the current implementation is that they are optionals. Since the image is based on a distroless image, mkdir isn't available so that's why I used copy.
> * Those folders are always empty on the CI so it's not really an issue for this kind of environment.
> 
> I do agree however that the content of the folders shouldn't be copied. If you can make it so the folders are still created by default without their content, that would be better.

Oh, oops, yes. I did mean to create them still, obviously. :)

comstud commented:
Thnx for the feedback. Please re-review. I've added the empty directory creations. That was just an oversight. While doing this, I also noticed my 130M golbat binary passed through to the docker daemon for no reason, since it is going to be rebuilt in there. So I added that to .dockerignore, also.

na-ji reviewed (approved):
Smart solution, thanks!

comstud commented:
np, ty!
